src/CanvasOptions.cpp

```diff
@@ -130,7 +130,7 @@ CanvasOptions::CanvasOptions( wxWindow *parent)
     wxStaticBoxSizer* boxNavMode = new wxStaticBoxSizer(new wxStaticBox(pDisplayPanel, wxID_ANY, _("Navigation Mode")), wxVERTICAL);
     generalSizer->Add(boxNavMode, 0, wxALL | wxEXPAND, border_size);
     
-    wxBoxSizer* rowOrientation = new wxBoxSizer(wxVERTICAL);
+    wxBoxSizer* rowOrientation = new wxBoxSizer(wxHORIZONTAL);
```

Comment:
@bdbcat in this case you are adding a wxBoxSizer pointer named rowOrientation to another wxStaticBoxSizer pointer named boxNavMode.  Because boxNavMode points to a sizer created with the wxVERTICAL flag, the wx library insists that when adding another sizer to it then it cannot have the same orientation flag (i.e. cannot be wxVERTICAL) so they throw a wxASSERT when the next line executes:

```boxNavMode->Add(rowOrientation);```

We want *rowOrientation to be populated horizontally anyway. In other words when you add the radio buttons to it, we want them side by side, not stacked one above the other.

To enforce their rules, the wx library overrides the wxVERTICAL flag and changes it to wxHORIZONTAL when they think the programmer has gotten it "wrong".  The library has always done this silently, but now they throw wxASSERTS in debug mode every time they override any incoming orientation flag.

And there is apparently a bug in wxWidgets/master that causes a crash in Windows debug mode every time one of these wxASSERTS happens.  I am working on that as well.

Anyway, this patch works for me, and does not generate wxASSERTS.  The behavior is otherwise unchanged because the library overrides the flags when they are "wrong".

P.S. One of the biggest perpetrators of this "error" was wxFormsBuilder.  They got it wrong for years which I guess is why the library silently corrects it in release mode.

Reply:
1. "the wx library insists that when adding another sizer to it then it cannot have the same orientation flag".  Can you point to the wx code for this?
2.  "We want *rowOrientation to be populated horizontally anyway."  No, not actually. We want them stacked, so that the overall window width is minimized, thus preserving more chart view.

Reply:
I understand your point.  Yes, we want them stacked vertically.  I will figure out how to get that without the wxASSERT.

Here is the code snippet from ./wxwidgets/src/common/sizer.cpp

```
//---------------------------------------------------------------------------
// wxBoxSizer
//---------------------------------------------------------------------------

wxSizerItem *wxBoxSizer::DoInsert(size_t index, wxSizerItem *item)
{
    const int flags = item->GetFlag();
    if ( IsVertical() )
    {
        wxASSERT_MSG
        (
            !(flags & wxALIGN_BOTTOM),
            wxS("Vertical alignment flags are ignored in vertical sizers")
        );

        // We need to accept wxALIGN_CENTRE_VERTICAL when it is combined with
        // wxALIGN_CENTRE_HORIZONTAL because this is known as wxALIGN_CENTRE
        // and we accept it historically in wxSizer API.
        if ( !(flags & wxALIGN_CENTRE_HORIZONTAL) )
        {
            wxASSERT_MSG <---------- it trips this assert 
            (
                !(flags & wxALIGN_CENTRE_VERTICAL),
                wxS("Vertical alignment flags are ignored in vertical sizers")
            );
        }
```

Put breakpoint here (about line 1140) in ./wxWidgets/src/common/appbase.cpp to catch the assert if/when it happens.
```
// default assert handler
static void
wxDefaultAssertHandler(const wxString& file,
                       int line,
                       const wxString& func,
                       const wxString& cond,
                       const wxString& msg)
{
```